spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/AbstractJacksonFieldSnippet.java

```diff
@@ -87,13 +90,37 @@ protected FieldDescriptors createFieldDescriptors(Operation operation,
 
     protected Type firstGenericType(MethodParameter param) {
         Type type = param.getGenericParameterType();
-        if (type instanceof ParameterizedType) {
+        if(type instanceof TypeVariable) {
+            TypeVariable tv = (TypeVariable)type;
+            return findTypeFromTypeVariable(tv, param.getContainingClass());
+        } else if (type instanceof ParameterizedType) {
+            ParameterizedType parameterizedType = (ParameterizedType) type;
+            Type actualArgument = parameterizedType.getActualTypeArguments()[0];
+            if(actualArgument instanceof Class) {
+                return actualArgument;
+            } else if(actualArgument instanceof TypeVariable) {
+                TypeVariable typeVariable = (TypeVariable)actualArgument;
+                return findTypeFromTypeVariable(typeVariable, param.getContainingClass());
+            }             
             return ((ParameterizedType) type).getActualTypeArguments()[0];
         } else {
             return Object.class;
         }
     }
 
+    protected Type findTypeFromTypeVariable(TypeVariable typeVariable, Class<?> clazz) {
+        Type defaultReturnValue = Object.class;
```

Comment:
This is only used at the bottom in the last return statement. I would recommend avoiding the variable completely and maybe add a comment. Alternatively, the variable could be moved to the return statement.

Reply:
Updated with better formatting